php/src/Google/Protobuf/Internal/GPBUtil.php

```diff
@@ -133,10 +132,10 @@ public static function checkInt64(&$var)
     public static function checkUint64(&$var)
     {
         if (is_numeric($var)) {
-            if (PHP_INT_SIZE == 8) {
-                $var = intval($var);
+            if (bccomp((string) $var, (string) PHP_INT_MAX) === 1) {
+                $var = (string) $var;
             } else {
-                $var = number_format($var, 0, ".", "");
```

Comment:
idk why we were doing this before, but this seems like a noticeably change on 32-bit platforms?  Could we limit this change to 64 bit?

Reply:
@mkruskal-google I don't really understand why it is used, seems useless, because any cast to string `(string) $value` provides the same result in any environment, but works much faster than `number_format` function made for printing numbers for humans. 

> but this seems like a noticeably change on 32-bit platforms
I think any which can fit to `int32` also should be converted to `int`. 

The PR isn't about that, but I think we can open an issue and refactor all the calls to casts.

Reply:
Yea I agree that your change seems better, but I'm not sure why we decided to format uint64 like this on 32-bit.  My point was that this changes the behavior in 32-bit by turning it into an int instead of a formatted string.  It just seems like an additional change that's out of scope for the original issue

Reply:
@mkruskal-google Aaah, yes, got it. I've reverted `number_format` too. I think we can write better tests and then do better refactoring.